src/lib/mdns/tests/TestServicePool.cpp

```diff
@@ -0,0 +1,117 @@
+#include "lib/mdns/ServicePool.h"
+
+#include <nlunit-test.h>
+
+#include "lib/support/CHIPMem.h"
+
+using chip::Mdns::ServicePool;
+
+void TestServicePool(nlTestSuite * suite, void * context)
```

Comment:
It would be nice to have more split-out tests, each testing one thing (like 'simple add/remove', 'remove missing', 'add multiple' etc.)